Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp

```diff
@@ -328,15 +328,30 @@ void GStreamerRegistryScanner::initializeDecoders(const GStreamerRegistryScanner
     bool shouldAddMP4Container = false;
 
     auto h264DecoderAvailable = factories.hasElementForMediaType(ElementFactories::Type::VideoDecoder, "video/x-h264, profile=(string){ constrained-baseline, baseline, high }", ElementFactories::CheckHardwareClassifier::Yes);
-    if (h264DecoderAvailable && (!m_isMediaSource || factories.hasElementForMediaType(ElementFactories::Type::VideoParser, "video/x-h264"))) {
+    auto h264AllFormatsDecoderAvailable = GStreamerRegistryScanner::RegistryLookupResult::merge(
+        factories.hasElementForMediaType(ElementFactories::Type::VideoDecoder, "video/x-h264, profile=(string){ constrained-baseline, baseline, high }, stream-format=(string)avc", ElementFactories::CheckHardwareClassifier::Yes),
+        factories.hasElementForMediaType(ElementFactories::Type::VideoDecoder, "video/x-h264, profile=(string){ constrained-baseline, baseline, high }, stream-format=(string)byte-stream", ElementFactories::CheckHardwareClassifier::Yes)
```

Comment:
Why filter on only a few profiles here?

Reply:
That was the case before, and I assume it's because you want to ensure the platform has a decoder that can at least handle those common profiles.

Reply:
Those are "the ones I know". The ones typically handled by avdec, for instance.